docs/api/field_type_reference.md

```diff
@@ -2240,17 +2240,20 @@ This Field Type validates and stores information about a user.
 |`login`|`string`|Username.|`john`|
 |`email`|`string`|The user's email address.|`john@smith.com`|
 |`passwordHash`|`string`|Hash of the user's password.|`1234567890abcdef`|
-|`passwordHashType`|`mixed`|Algorithm user for generating password hash as a `PASSWORD_HASH_*` constant defined in `eZ\Publish\Core\Repository\Values\User\User` class.|`User::PASSWORD_HASH_MD5_USER`|
+|`passwordHashType`|`mixed`|Algorithm user for generating password hash as a `PASSWORD_HASH_*` constant defined in `eZ\Publish\Core\Repository\Values\User\User` class.|`User::PASSWORD_HASH_PHP_DEFAULT`|
 |`maxLogin`|`int`|Maximum number of concurrent logins.|`1000`|
 
 ###### Available password hash types
 
 |Constant|Description|
 |------|------|
-|`eZ\Publish\Core\Repository\Values\User\User::PASSWORD_HASH_MD5_PASSWORD`|MD5 hash of the password, not recommended.|
-|`eZ\Publish\Core\Repository\Values\User\User::PASSWORD_HASH_MD5_USER`|MD5 hash of the password and username.|
-|`eZ\Publish\Core\Repository\Values\User\User::PASSWORD_HASH_MD5_SITE`|MD5 hash of the password, username and site name.|
-|`eZ\Publish\Core\Repository\Values\User\User::PASSWORD_HASH_PLAINTEXT`|Passwords are stored in plaintext, should not be used for real sites.|
+|`eZ\Publish\Core\Repository\Values\User\User::DEFAULT_PASSWORD_HASH`|Default password hash, used when none is specified, may change over time.|
+|`eZ\Publish\Core\Repository\Values\User\User::PASSWORD_HASH_PHP_DEFAULT`|Passwords hashed by PHP's default algorithm, which may change over time.|
+|`eZ\Publish\Core\Repository\Values\User\User::PASSWORD_HASH_BCRYPT`|Bcrypt hash of the password.|
+|`eZ\Publish\Core\Repository\Values\User\User::PASSWORD_HASH_MD5_PASSWORD`|MD5 hash of the password, not recommended. Deprecated in v1.13, removed in v3.0.|
+|`eZ\Publish\Core\Repository\Values\User\User::PASSWORD_HASH_MD5_USER`|MD5 hash of the password and username. Deprecated in v1.13, removed in v3.0.|
+|`eZ\Publish\Core\Repository\Values\User\User::PASSWORD_HASH_MD5_SITE`|MD5 hash of the password, username and site name. Deprecated in v1.13, removed in v3.0.|
+|`eZ\Publish\Core\Repository\Values\User\User::PASSWORD_HASH_PLAINTEXT`|Passwords are stored in plaintext, should not be used for real sites. Deprecated in v1.13, removed in v3.0.|
```

Comment:
```suggestion
|`eZ\Publish\Core\Repository\Values\User\User::PASSWORD_HASH_PLAINTEXT`|Passwords are stored in plaintext, should not be used for real sites. Deprecated.|
```
Here and in the two cases above we can jut remove the line in branches for v3

Reply:
Completely remove the lines in v3 doc, agree.
I'd say it can useful for a v1 / v2 dev to know when exactly these were deprecated, and it's common practice in the code itself to include it.

Reply:
It might even be wise to add an alert box here, to make it very clear that using these deprecated hash types is a security risk.

Reply:
Added the warning box in new commit.